### PR TITLE
Make the creative mod hand dig 'dig_immediate' nodes fast

### LINKED ISSUE #2427: dig_immediate = 2 nodes dig slowly in creative mode
I'm using creative mod in a simple game of my own.
I have some nodes that have group `dig_immediate = 2`, with or without other dig groups, to be diggable by hand at moderate speed. In creative mode these nodes dig slowly.

To reproduce in MTG:
Alter sapling to change `dig_immediate' from 3 to 2:
```
	groups = {snappy = 2, dig_immediate = 2, flammable = 2,
		attached_node = 1, sapling = 1},
```
In creative mode, try digging a row of saplings.

I suspect this hasn't been noticed in MTG yet because the only `dig_immediate` nodes are value 3 which digs very fast.

Adding `dig_immediate = caps` to the hand redefinition doesn't seem to help.

Hmmm but then, 'dig immediate' suggests that only the value of 3 should be used to make it immediate. Using 'oddly breakable by hand' seems wrong for nodes that make sense to be diggable by hand.

### PR DESCRIPTION
**Pull request by paramat**
Closes #2427 
~Pre-approved by sfan5 in the issue.~
Course of action detailed in the issue and in https://github.com/minetest/minetest/pull/8723

### CODE REVIEW

**Comment by sfan5** [comment]
> Pre-approved by sfan5 in the issue.

No, I "approved" the course of action.
This approval does not extend to a specific code implementation, such as this one which is by the way not correct (thought its marked as WIP).

**Comment by paramat** [comment]
Ok, if this is incorrect then indeed it isn't approved. I just added that because i assumed the code was trivial and obvious and would work, haven't tested yet.
So, please could you help by explaining why it isn't correct?

**Comment by paramat** [comment]
Hmmm it seems to work ... ?

**Comment by sfan5** [comment]
> So, please could you help by explaining why it isn't correct?

dig_immediate = 1 does not have a meaning so we should probably not define a digtime, minor issue
dig_immediate = 2 is supposed to be instant to dig, it will not be instant with this code
dig_immediate = 3 will indeed work fine like this, but when you accommodate the rest, you need to it differently

as I've hinted in the engine PR description:
> minetest_game can then use `dig_immediate = {times = {[2]=42/256, [3]=0}, maxlevel = 3},`
to make dig_immediate=2 nodes dig about as fast as everything else

**Comment by paramat** [comment]
Oh sorry, i missed the information in the engine PR.
And yes i just realised it's not like the other groups.
Thanks.

**Comment by sfan5** [comment]
> I'm really not too good with tool stuff, i should have left this task to you =(

Me neither, I just happened to know since I <i>just</i> read the code to make the engine PR.

File: mods/creative/init.lua [hunk 1]
```diff
@@ -54,6 +54,7 @@ if creative_mode_cache then
 				snappy  = caps,
 				choppy  = caps,
 				oddly_breakable_by_hand = caps,
+				dig_immediate = caps,
```
Comment (DS-Minetest) [inline mods/creative/init.lua]:
```suggestion
				dig_immediate = {times = {nil, digtime, 0}, uses = 0, maxlevel = 256},
```
(Copied and modified from line 41.)

Comment (sfan5) [reply mods/creative/init.lua]: I'd prefer `times = {[2] = digtime, [3] = 0}` because its more explicit

**Comment by paramat** [comment]
Ok i'll update this on your suggestions, thanks.

**Comment by paramat** [comment]
Updated.

**Pull request merged**